src/goto-instrument/goto_instrument_parse_options.cpp

```diff
@@ -898,11 +898,13 @@ void goto_instrument_parse_optionst::get_goto_program()
 {
   status() << "Reading GOTO program from `" << cmdline.args[0] << "'" << eom;
 
+  config.set(cmdline);
+
   if(read_goto_binary(cmdline.args[0],
     goto_model, get_message_handler()))
     throw 0;
 
-  config.set(cmdline);
+  config.set_from_symbol_table(goto_model.symbol_table);
```

Comment:
We might need a custom wrapper script to test this, but some tests certainly seem necessary. In particular, the command-line should actually be ignored.